app/src/main/java/org/p2p/wallet/receive/list/TokenListPresenter.kt

```diff
@@ -48,8 +47,12 @@ class TokenListPresenter(
 
     private fun observeTokens() {
         launch {
-            interactor.getTokenListFlow().collect { tokens ->
-                view?.showItems(tokens, scrollToUp)
+            interactor.getTokenListFlow().collect { data ->
+                if (data.result.isNotEmpty()) {
+                    view?.showItems(data.result, scrollToUp)
```

Comment:
have you tried resolve such things using sealed ViewState class? only one func in view - renderViewState